tests/unit/controlgroup/controlgroup_methods.js

```diff
@@ -0,0 +1,130 @@
+module( "Controlgroup: methods" );
+
+test( "destroy", function() {
+	expect( 1 );
+	domEqual( ".controlgroup", function() {
+		$( ".controlgroup" ).controlgroup().controlgroup( "destroy" );
+	});
+});
+test( "disable", function() {
+	var tests = 1,
+		element = $( ".controlgroup" ).controlgroup().controlgroup( "disable" );
+	strictEqual( element.hasClass( "ui-state-disabled" ), false,
+		"ui-state-disabled is not present on widget after disabling" );
+	$.each( $.ui.controlgroup.prototype.options.items, function( widget, selector ){
+		element.children( selector ).each(function(){
+			expect( ++tests );
+			strictEqual( $( this )[ widget ]( "widget" ).hasClass( "ui-state-disabled"), true,
+				"Child " + widget + " widgets are disabled" );
+		});
+	});
+});
+
+test( "enable", function() {
+	var tests = 1,
+		element = $( ".controlgroup" ).controlgroup().controlgroup( "enable" );
+	strictEqual( element.hasClass( "ui-state-disabled" ), false,
+		"ui-state-disabled is not present on widget after enabling" );
+	$.each( $.ui.controlgroup.prototype.options.items, function( widget, selector ){
+		element.children( selector ).each(function(){
+			expect( ++tests );
+			strictEqual( $( this )[ widget ]( "widget" ).hasClass( "ui-state-disabled"), false,
+				"Child " + widget + " widgets are not disabled" );
+		});
+	});
+});
+
+function hasCornerClass( className, element ) {
+	if ( className ) {
+		return element.hasClass( className );
+	}
+
+	return element.attr( "class" ).match( /ui-corner/g );
+}
+test( "refresh", function() {
```

Comment:
This test has way too much logic in it. I'm not going into the details for now, let's improve the above tests first, that may provide some ideas for dealing with this one.